### PR TITLE
Java Solution for Subarray Sums II

### PR DESCRIPTION
**Pull request by brad182**
removed spoiler because it isn't necessary

### CODE REVIEW

File: solutions/cses-1661.mdx [hunk 1]
```diff
@@ -2,24 +2,27 @@
 id: cses-1661
 source: CSES
 title: Subarray Sums II
-author: Qi Wang
+author: Qi Wang, Brad Ma
 ---
 
 # Problem
```
Comment (HarryWangATX) [inline solutions/cses-1661.mdx:8]:
```suggestion
## Problem
```

File: solutions/cses-1661.mdx [hunk 2]
```diff
@@ -2,24 +2,27 @@
...
 
 We are asked to find the number of subarrays that sum up to $x$ given the size
 of the array and its elements.
 
-<Spoiler title="Solution">
-
 We can have a map that keeps track of the prefix sums. At each index $i$, we can
```
Comment (HarryWangATX) [inline solutions/cses-1661.mdx:13]:
```suggestion
## Explanation

We can have a map that keeps track of the prefix sums. At each index $i$, we can
```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
just some quick fixes

File: solutions/cses-1661.mdx [hunk 3]
```diff
@@ -61,5 +66,57 @@ int main() {
...
+			 * If there is a subarray with a prefix sum of prefix_sum - X,
+			 * we can exclude it from our current subarray to get the desired sum.
+			 * Thus, we can add the number of those subarrays to our answer.
+			 */
+			if (sums.containsKey(prefixSum - target)) {  // check if it is in our map
+				answer += sums.get(prefixSum - target);
+			}
+
+			// Increment the amount of prefix sums with a sum of prefix_sum
+			if (! sums.containsKey(prefixSum)) {  // not yet in map, so add it
```
Comment (nwatx) [inline solutions/cses-1661.mdx:106]:
```suggestion
			if (!sums.containsKey(prefixSum)) {  // not yet in map, so add it
```

File: solutions/cses-1661.mdx [hunk 4]
```diff
@@ -61,5 +66,57 @@ int main() {
...
+		int[] array = new int[arraySize];
+
+		for (int x = 0; x < arraySize; x++) {
+			array[x] = io.nextInt();
+		}
+		
+		long prefixSum = 0;
+		long answer = 0;
+		Map<Long, Integer> sums = new HashMap<Long, Integer>();
+		sums.put((long) 0, 1);
```
Comment (nwatx) [inline solutions/cses-1661.mdx:92]: is `(long)` necessary

File: solutions/cses-1661.mdx [hunk 5]
```diff
@@ -2,24 +2,29 @@
...
+## Problem
 
 We are asked to find the number of subarrays that sum up to $x$ given the size
 of the array and its elements.
 
-<Spoiler title="Solution">
+## Explanation
 
 We can have a map that keeps track of the prefix sums. At each index $i$, we can
 count the number of prefixes with sum equal to $\texttt{prefixSum}[i] - x$. This
```
Comment (nwatx) [inline solutions/cses-1661.mdx:16]: I feel like using the word prefix isn't specific enough. Maybe do it in terms of \sum{0...i} or something

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/cses-1661.mdx (hunk 4 above)
Comment (brad182) [reply solutions/cses-1661.mdx:92]: yes according to my compiler, it always grays out if it isn't necessary but this time it is necessary and didn't gray out

File: solutions/cses-1661.mdx (hunk 5 above)
Comment (brad182) [reply solutions/cses-1661.mdx:16]: lol ask harry for that i just wrote the java solution

**Review by dongliu0426 — APPROVED** [review]

**Review by maggie-j-liu — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
looks good 👍

**Review by ar492 — APPROVED** [review]

**Review by nathangong — APPROVED** [review]

**Review by nwatx — APPROVED** [review]

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
just one little thing

**Comment by brad182** [comment]
6 checks and no merge still?

**Pull request merged**